arbos/internal_tx.go

```diff
@@ -107,7 +107,12 @@ func ApplyInternalTxUpdate(tx *types.ArbitrumInternalTx, state *arbosState.Arbos
 		_ = state.RetryableState().TryToReapOneRetryable(currentTime, evm, util.TracingDuringEVM)
 		_ = state.RetryableState().TryToReapOneRetryable(currentTime, evm, util.TracingDuringEVM)
 
-		state.L2PricingState().UpdatePricingModel(l2BaseFee, timePassed, false)
+		// For ArbOS version 50 and above call Multi-Constraint Pricer update
+		if state.ArbOSVersion() >= params.ArbosVersion_50 {
+			state.L2PricingState().UpdatePricingModelMultiConstraints(timePassed)
```

Comment:
To be consistent with AddToGasPool, `UpdatePricingModel` should receive the ArbOS version and make this check inside.